safep/fepruns.py

```
@@ -60,26 +61,99 @@ class FepRun:
+
+        if self.per_lambda_convergence is None:
+            self.per_lambda_convergence = safep.do_per_lambda_convergence(self.u_nk)
+
+        for field in fields(self):
+            attr = getattr(self, field.name)
+            if not isinstance(attr, field.type):
+                if field.type == pd.DataFrame:
+                    setattr(self, field.name, pd.DataFrame(attr))
+                elif field.type == np.ndarray:
+                    setattr(self, field.name, np.asarray(attr))
+                elif field.type == str:
```

Ruff (0.15.21)

[error] Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)